src/NServiceBus.AmazonSQS/Batcher.cs

```diff
@@ -0,0 +1,57 @@
+namespace NServiceBus.Transports.SQS
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using AmazonSQS;
+
+    static class Batcher
+    {
+        public static IReadOnlyList<BatchEntry> Batch(IEnumerable<PreparedMessage> preparedMessages)
+        {
+            var allBatches = new List<BatchEntry>();
+            var currentDestinationBatches = new Dictionary<string, PreparedMessage>();
+
+            var groupByDestination = preparedMessages.GroupBy(m => m.QueueUrl, StringComparer.OrdinalIgnoreCase);
```

Comment:
given that SQS queue names and URLs are case-sensitive (IIRC) does this have the risk of putting in the same batch messages for different destinations? e.g. `Sales` and `sales` will be in the same group.

Reply:
Great catch. I wanted to use Ordinal only and not ignorecase